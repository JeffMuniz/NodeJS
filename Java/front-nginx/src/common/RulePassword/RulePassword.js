import React from "react";
import { Icon } from "@common";
import { string } from "prop-types";

import { LabelSucess, LabelError } from "./RulePassword.styles";

const RulePassword = ({ text, value, error, rule, id }) => {
  const result = RegExp(rule).test(value);

  return result ? (
    <LabelSucess id={id}>
      {text} <Icon name="check-input" />
    </LabelSucess>
  ) : (
    <LabelError error={error} id={id}>
      {text} {error ? <Icon name="nocheck-input" /> : ""}
    </LabelError>
  );
};
export default RulePassword;

RulePassword.propTypes = {
  text: string.isRequired,
  value: string.isRequired,
  error: string.isRequired,
  rule: string.isRequired,
  id: string,
};

RulePassword.defaultProps = {
  id: "",
};
