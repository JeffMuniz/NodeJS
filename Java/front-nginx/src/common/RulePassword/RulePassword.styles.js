import styled from "styled-components";
import { red, veryLigtherGrey, green } from "@colors";

const BaseListItem = styled.li`
  margin-right: 35px;
  line-height: 22px;
  font-size: 14px;
  letter-spacing: -0.3px;
  text-transform: lowercase;
  font-style: normal;
`;

export const LabelSucess = styled(BaseListItem)`
  color: ${green};
`;

export const LabelError = styled(BaseListItem)`
  color: ${({ error }) => (error ? red : veryLigtherGrey)};
`;
