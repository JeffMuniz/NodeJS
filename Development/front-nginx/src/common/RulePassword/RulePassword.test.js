import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import RulePassword from "./RulePassword";

const props = {
  text: "Teste",
  value: "Aser",
  rule: "/^(?=.*?[A-Z])/",
};

jest.mock("@common", () => ({
  Icon: "Icon",
}));

describe("Rule Password", () => {
  it("Render Rule Password", () => {
    const renderer = new ShallowRenderer();
    const defaultProps = {
      ...props,
      error: false,
    };

    // when
    renderer.render(<RulePassword {...defaultProps} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });

  it("Render error", () => {
    const defaultProps = {
      ...props,
      error: true,
    };
    const renderer = new ShallowRenderer();

    // when
    renderer.render(<RulePassword {...defaultProps} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });
});
