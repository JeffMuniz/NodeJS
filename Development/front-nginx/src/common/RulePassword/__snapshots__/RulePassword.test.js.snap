// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rule Password Render Rule Password 1`] = `
<Styled(styled.li)
  error={false}
  id=""
>
  Teste
   
  
</Styled(styled.li)>
`;

exports[`Rule Password Render error 1`] = `
<Styled(styled.li)
  error={true}
  id=""
>
  Teste
   
  <Icon
    name="nocheck-input"
  />
</Styled(styled.li)>
`;
